include/alpaka/mem/buf/cpu/Copy.hpp

```diff
@@ -39,29 +39,6 @@ namespace alpaka
             using SrcSize = Idx<TViewSrc>;
             using Elem = alpaka::Elem<TViewSrc>;
 
-            static_assert(!std::is_const_v<TViewDst>, "The destination view can not be const!");
-
-            static_assert(
-                Dim<TViewDst>::value == Dim<TViewSrc>::value,
-                "The source and the destination view are required to have the same dimensionality!");
-            static_assert(
-                Dim<TViewDst>::value == Dim<TExtent>::value,
-                "The views and the extent are required to have the same dimensionality!");
-            static_assert(
-                Dim<TViewDst>::value == TDim::value,
-                "The destination view and the input TDim are required to have the same dimensionality!");
```

Comment:
Asserts of this sort are not replicated in the traits, because we literally pass `Dim<TViewDst>` as template argument for the `TDim` parameter.